suppress FakeItEasy.Tests "obsolete" warnings

There are currently 16 warnings, because we test methods that are marked obsolete (just because they're obsolete doesn't mean we want to break them without knowing):

```
FakeItEasy.Tests\AnyTests.cs(12,25,12,28): warning CS0618: 'FakeItEasy.Any' is obsolete: 'Methods provided by this class are now available directly on the A-class.'
FakeItEasy.Tests\AnyTests.cs(20,25,20,28): warning CS0618: 'FakeItEasy.Any' is obsolete: 'Methods provided by this class are now available directly on the A-class.'
FakeItEasy.Tests\Creation\DefaultFakeObjectCreatorTests.cs(121,13,121,16): warning CS0618: 'FakeItEasy.Any' is obsolete: 'Methods provided by this class are now available directly on the A-class.'
FakeItEasy.Tests\Creation\DefaultFakeObjectCreatorTests.cs(121,13,121,37): warning CS0618: 'FakeItEasy.Any.CallTo(object)' is obsolete: 'Use the CallTo-method on the A-class instead.'
FakeItEasy.Tests\Core\DelegateFakeObjectContainerTests.cs(59,13,59,16): warning CS0618: 'FakeItEasy.Any' is obsolete: 'Methods provided by this class are now available directly on the A-class.'
FakeItEasy.Tests\Core\DelegateFakeObjectContainerTests.cs(59,13,59,29): warning CS0618: 'FakeItEasy.Any.CallTo(object)' is obsolete: 'Use the CallTo-method on the A-class instead.'
FakeItEasy.Tests\Expressions\ExpressionArgumentConstraintFactoryTests.cs(36,13,36,16): warning CS0618: 'FakeItEasy.Any' is obsolete: 'Methods provided by this class are now available directly on the A-class.'
FakeItEasy.Tests\Expressions\ExpressionArgumentConstraintFactoryTests.cs(36,13,36,37): warning CS0618: 'FakeItEasy.Any.CallTo(object)' is obsolete: 'Use the CallTo-method on the A-class instead.'
FakeItEasy.Tests\Core\NullFakeObjectContainerTests.cs(34,13,34,16): warning CS0618: 'FakeItEasy.Any' is obsolete: 'Methods provided by this class are now available directly on the A-class.'
FakeItEasy.Tests\Core\NullFakeObjectContainerTests.cs(34,13,34,29): warning CS0618: 'FakeItEasy.Any.CallTo(object)' is obsolete: 'Use the CallTo-method on the A-class instead.'
FakeItEasy.Tests\Configuration\RuleBuilderTests.cs(191,13,191,16): warning CS0618: 'FakeItEasy.Any' is obsolete: 'Methods provided by this class are now available directly on the A-class.'
FakeItEasy.Tests\Configuration\RuleBuilderTests.cs(191,13,191,29): warning CS0618: 'FakeItEasy.Any.CallTo(object)' is obsolete: 'Use the CallTo-method on the A-class instead.'
FakeItEasy.Tests\Core\FakeScopeTests.cs(86,21,86,24): warning CS0618: 'FakeItEasy.Any' is obsolete: 'Methods provided by this class are now available directly on the A-class.'
FakeItEasy.Tests\Core\FakeScopeTests.cs(86,21,86,37): warning CS0618: 'FakeItEasy.Any.CallTo(object)' is obsolete: 'Use the CallTo-method on the A-class instead.'
FakeItEasy.Tests\SyntaxTests.cs(14,13,14,16): warning CS0618: 'FakeItEasy.Any' is obsolete: 'Methods provided by this class are now available directly on the A-class.'
FakeItEasy.Tests\SyntaxTests.cs(14,13,14,28): warning CS0618: 'FakeItEasy.Any.CallTo(object)' is obsolete: 'Use the CallTo-method on the A-class instead.'
```

If we ignore these (and I'd propose on case-by-case basis, not on the project as a whole), it'll clean up the build output a little.

Then we should be able to turn on "Treat warnings as errors", like the other projects.

--- a/Source/FakeItEasy.Tests/AnyTests.cs
+++ b/Source/FakeItEasy.Tests/AnyTests.cs
@@ -6,6 +6,7 @@ namespace FakeItEasy.Tests
     public class AnyTests
         : ConfigurableServiceLocatorTestBase
     {
+#pragma warning disable 618
         [Test]
         public void Static_equals_delegates_to_static_method_on_object()
         {
@@ -19,6 +20,7 @@ namespace FakeItEasy.Tests
 
             Assert.That(Any.ReferenceEquals(s, s), Is.True);
         }
+#pragma warning restore 618
 
         protected override void OnSetup()
         {

--- a/Source/FakeItEasy.Tests/Configuration/RuleBuilderTests.cs
+++ b/Source/FakeItEasy.Tests/Configuration/RuleBuilderTests.cs
@@ -188,7 +188,10 @@
             this.builder.DoesNothing();
 
             var call = A.Fake<IInterceptedFakeObjectCall>();
+
+#pragma warning disable 618
             Any.CallTo(call).Throws(new AssertionException("Applicator should do nothing."));
+#pragma warning restore 618
 
             this.builder.RuleBeingBuilt.Applicator(call);
         }

--- a/Source/FakeItEasy.Tests/Core/DelegateFakeObjectContainerTests.cs
+++ b/Source/FakeItEasy.Tests/Core/DelegateFakeObjectContainerTests.cs
@@ -56,7 +56,10 @@
         public void ConfigureFakeObject_should_do_nothing()
         {
             var fake = A.Fake<IFoo>();
+
+#pragma warning disable 618
             Any.CallTo(fake).Throws(new InvalidOperationException());
+#pragma warning restore 618
 
             var container = this.CreateContainer();
 

--- a/Source/FakeItEasy.Tests/Core/FakeScopeTests.cs
+++ b/Source/FakeItEasy.Tests/Core/FakeScopeTests.cs
@@ -83,7 +83,9 @@
             {
                 using (Fake.CreateScope())
                 {
+#pragma warning disable 618
                     Any.CallTo(fake).Throws(new InvalidOperationException());
+#pragma warning restore 618
                 }
 
                 fake.Bar();

--- a/Source/FakeItEasy.Tests/Core/NullFakeObjectContainerTests.cs
+++ b/Source/FakeItEasy.Tests/Core/NullFakeObjectContainerTests.cs
@@ -31,7 +31,10 @@
         public void ConfigureFakeObject_should_do_nothing()
         {
             var fake = A.Fake<IFoo>();
+
+#pragma warning disable 618
             Any.CallTo(fake).Throws(new InvalidOperationException());
+#pragma warning restore 618
 
             var container = new NullFakeObjectContainer();
 

--- a/Source/FakeItEasy.Tests/Creation/DefaultFakeObjectCreatorTests.cs
+++ b/Source/FakeItEasy.Tests/Creation/DefaultFakeObjectCreatorTests.cs
@@ -118,7 +118,9 @@ namespace FakeItEasy.Tests.Creation
             this.fakeObjectCreator.CreateFake(typeof(IFoo), FakeOptions.Empty, A.Dummy<IDummyValueCreationSession>(), throwOnFailure: false);
 
             // Assert
+#pragma warning disable 618
             Any.CallTo(this.thrower).MustNotHaveHappened();
+#pragma warning restore 618
         }
 
         [Test]

--- a/Source/FakeItEasy.Tests/Expressions/ExpressionArgumentConstraintFactoryTests.cs
+++ b/Source/FakeItEasy.Tests/Expressions/ExpressionArgumentConstraintFactoryTests.cs
@@ -33,7 +33,10 @@
         {
             // Arrange
             var constraint = A.Dummy<IArgumentConstraint>();
+
+#pragma warning disable 618
             Any.CallTo(this.trapper).WithReturnType<IEnumerable<IArgumentConstraint>>().Returns(new[] { constraint });
+#pragma warning restore 618
 
             // Act
             var result = this.factory.GetArgumentConstraint(BuilderForParsedArgumentExpression.BuildWithDefaults());

--- a/Source/FakeItEasy.Tests/FakeItEasy.Tests.csproj
+++ b/Source/FakeItEasy.Tests/FakeItEasy.Tests.csproj
@@ -58,7 +58,6 @@
     <WarningLevel>4</WarningLevel>
     <CodeAnalysisRuleSet>..\FakeItEasy.Tests.ruleset</CodeAnalysisRuleSet>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
-    <WarningsNotAsErrors>618</WarningsNotAsErrors>
     <StyleCopTreatErrorsAsWarnings>false</StyleCopTreatErrorsAsWarnings>
     <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>

--- a/Source/FakeItEasy.Tests/SyntaxTests.cs
+++ b/Source/FakeItEasy.Tests/SyntaxTests.cs
@@ -11,7 +11,9 @@
         {
             var foo = A.Fake<Foo>();
 
+#pragma warning disable 618
             Any.CallTo(foo).Throws(new ArgumentNullException());
+#pragma warning restore 618
 
             A.CallTo(() => foo.Bar())
                 .Returns("1");

